templates/repo/settings/secrets.tmpl

```diff
@@ -0,0 +1,60 @@
+<div class="ui container">
+		<h4 class="ui top attached header">
+			{{.locale.Tr "secrets.secrets"}}
+			<div class="ui right">
+				<div class="ui primary tiny show-panel button" data-panel="#add-secret-panel">{{.locale.Tr "secrets.creation"}}</div>
+			</div>
+		</h4>
+		<div class="ui attached segment">
+			<div class="{{if not .HasError}}hide {{end}}mb-4" id="add-secret-panel">
+				<form class="ui form" action="{{.Link}}?act=secret" method="post">
+					{{.CsrfTokenHtml}}
+					<div class="field">
+						{{.locale.Tr "secrets.description"}}
+					</div>
+					<div class="field {{if .Err_Title}}error{{end}}">
+						<label for="title">{{.locale.Tr "secrets.name"}}</label>
+						<input id="ssh-key-title" name="title" value="{{.title}}" autofocus required pattern="^[a-zA-Z_][a-zA-Z0-9_.-]*$">
+					</div>
+					<div class="field {{if .Err_Content}}error{{end}}">
+						<label for="content">{{.locale.Tr "secrets.value"}}</label>
+						<textarea id="ssh-key-content" name="content" required placeholder="{{.locale.Tr "secrets.creation.value_placeholder"}}">{{.content}}</textarea>
+					</div>
+					<button class="ui green button">
+						{{.locale.Tr "secrets.creation"}}
+					</button>
+					<button class="ui hide-panel button" data-panel="#add-secret-panel">
+						{{.locale.Tr "cancel"}}
+					</button>
+				</form>
+			</div>
+			{{if .Secrets}}
+				<div class="ui key list">
+					{{range .Secrets}}
+						<div class="item">
+							<div class="right floated content">
+								<button class="ui red tiny button delete-button" data-url="{{$.Link}}/delete?act=secret" data-id="{{.ID}}">
+									{{$.locale.Tr "settings.delete_key"}}
+								</button>
+							</div>
+							<div class="left floated content">
+								<i>{{svg "octicon-key" 32}}</i>
+							</div>
+							<div class="content">
+								<strong>{{.Name}}</strong>
+								<div class="print meta">******</div>
```

Comment:
You got rid of that helper function?

Reply:
Given that there is more than one instance where we use it, it is probably still a good idea to add it.

Reply:
Are you sure?

😂 I can even say I didn't remove the helper function, just renamed it from `Shadow` to `******`, they have the same return value and the same length of name.

I'm kidding, but you know what I mean.

Reply:
Looks like the logic is the same. Since we don't plan display the real value in the frontend, just asterisk here looks like reasonable.